charts/descheduler/templates/cronjob.yaml

```
@@ -33,6 +33,9 @@ spec:
             {{- .Values.podLabels | toYaml | nindent 12 }}
             {{- end }}
         spec:
+          {{- if .Values.nodeSelector }}
+          nodeSelector: {{ .Values.nodeSelector }}
+          {{- end }}
```

I don't think this would work. I'd suggest we use the idiomatic way that Helm sets these up for new charts.

```suggestion
          {{- with .Values.nodeSelector }}
          nodeSelector:
            {{- toYaml . | nindent 12 }}
          {{- end }}
```